### PR TITLE
feat: include rich passage stylesheet to compiled delivery data

### PR DESCRIPTION
**Pull request by pnal**
How to test:
- Create or import a test containing styled passages.
- Publish a local delivery
- Login as a test taker 
- Check that styles are applied

The functionality was tested on CG test runner, we still need to deploy a new integration environment to test remote publishing and NextGen test runner.

### CODE REVIEW

File: model/compile/QtiAssetCompiler/QtiItemAssetCompiler.php [hunk 1]
```diff
@@ -145,4 +148,9 @@ private function getQtiItemAssetReplacer(): QtiItemAssetReplacer
     {
         return $this->getServiceLocator()->get(QtiItemAssetReplacer::SERVICE_ID);
     }
+
+    private function getXIncludeAdditionalAssetInjector(): XIncludeAdditionalAssetInjector
+    {
+        return $this->getServiceLocator()->get(XIncludeAdditionalAssetInjector::class);
```
Comment (mike-ionut-mihai-sandu-tao) [inline model/compile/QtiAssetCompiler/QtiItemAssetCompiler.php:154]: should we use the SERVICE_ID?

File: model/compile/QtiAssetCompiler/XIncludeAdditionalAssetInjector.php [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+use oat\taoQtiItem\model\pack\QtiAssetPacker\PackedAsset;
+use oat\taoQtiItem\model\qti\exception\QtiModelException;
+use oat\taoQtiItem\model\qti\Item;
+use oat\taoQtiItem\model\qti\Stylesheet;
+use Psr\Http\Message\StreamInterface;
+
+class XIncludeAdditionalAssetInjector extends ConfigurableService
+{
+    private const COMPILED_PASSAGE_STYLESHEET_FILENAME_PREFIX = 'passage';
+
```
Comment (mike-ionut-mihai-sandu-tao) [inline model/compile/QtiAssetCompiler/XIncludeAdditionalAssetInjector.php:39]: missing SERVICE_ID

File: model/compile/QtiAssetCompiler/XIncludeAdditionalAssetInjector.php [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+    /**
+     * @throws common_Exception
+     */
+    public function injectNonRDFXincludeRelatedAssets(
+        Item $qtiItem,
+        Directory $publicDirectory,
+        PackedAsset $packedAsset
+    ): void {
+        if ($packedAsset->getType() !== 'xinclude') {
```
Comment (mike-ionut-mihai-sandu-tao) [inline model/compile/QtiAssetCompiler/XIncludeAdditionalAssetInjector.php:48]: can we extract into a const?

Comment (mike-ionut-mihai-sandu-tao) [inline model/compile/QtiAssetCompiler/XIncludeAdditionalAssetInjector.php:48]: can we extract this check into a different function?

File: model/compile/QtiAssetCompiler/XIncludeAdditionalAssetInjector.php [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+     * @throws QtiModelException
+     * @throws FileExistsException
+     * @throws common_Exception
+     */
+    private function includeSharedStimulusStylesheet(
+        Item $qtiItem,
+        Directory $publicDirectory,
+        StreamInterface $stylesheetContent
+    ): void {
+        $stylesheetUrl = uniqid(
```
Comment (mike-ionut-mihai-sandu-tao) [inline model/compile/QtiAssetCompiler/XIncludeAdditionalAssetInjector.php:81]: let's use uuid and not uniqid
and for that uuid we have a trait that returns a uuid4

File: model/compile/QtiAssetCompiler/XIncludeAdditionalAssetInjector.php [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+                'title' => AssetStylesheetLoader::ASSET_CSS_FILENAME,
+                'type' => 'text/css'
+            ]
+        );
+
+        $qtiItem->addStylesheet($qtiStylesheet);
+    }
+
+    /**
+     * @return AssetStylesheetLoader
```
Comment (mike-ionut-mihai-sandu-tao) [inline model/compile/QtiAssetCompiler/XIncludeAdditionalAssetInjector.php:97]: not required as we have the return strong type already

File: test/unit/model/compile/XIncludeAdditionalAssetInjectorTest.php [hunk 6]
```diff
@@ -0,0 +1,162 @@
...
+
+        return $this->createConfiguredMock(
+            PackedAsset::class,
+            [
+                'getMediaAsset' => $mediaAsset,
+                'getType' => $type
+            ]
+        );
+    }
+}
```
Comment (mike-ionut-mihai-sandu-tao) [inline test/unit/model/compile/XIncludeAdditionalAssetInjectorTest.php:162]: \n

**Review by mike-ionut-mihai-sandu-tao — COMMENTED** [review]
looks ok , just a few comments

File: model/compile/QtiAssetCompiler/XIncludeAdditionalAssetInjector.php (hunk 2 above)
Comment (bartlomiejmarszal) [reply model/compile/QtiAssetCompiler/XIncludeAdditionalAssetInjector.php:44]: I don't think we need to include SERVICE_ID in this configurable service.

File: model/compile/QtiAssetCompiler/XIncludeAdditionalAssetInjector.php [hunk 7]
```diff
@@ -0,0 +1,111 @@
...
+     * @throws FileExistsException
+     * @throws common_Exception
+     */
+    private function includeSharedStimulusStylesheet(
+        Item $qtiItem,
+        Directory $publicDirectory,
+        StreamInterface $stylesheetContent
+    ): void {
+        $stylesheetUrl = $this->getUniquePrimaryKey(
+            ) . self::COMPILED_PASSAGE_STYLESHEET_FILENAME_PREFIX . AssetStylesheetLoader::ASSET_CSS_FILENAME;
```
Comment (bartlomiejmarszal) [inline model/compile/QtiAssetCompiler/XIncludeAdditionalAssetInjector.php:87]: This is weird line break, no?

File: model/compile/QtiAssetCompiler/XIncludeAdditionalAssetInjector.php [hunk 8]
```diff
@@ -0,0 +1,111 @@
...
+use oat\taoQtiItem\model\qti\Stylesheet;
+use Psr\Http\Message\StreamInterface;
+
+class XIncludeAdditionalAssetInjector extends ConfigurableService
+{
+    use UuidPrimaryKeyTrait;
+
+    public const XINCLUDE_ASSET_TYPE = 'xinclude';
+
+    private const COMPILED_PASSAGE_STYLESHEET_FILENAME_PREFIX = 'passage';
```
Comment (bartlomiejmarszal) [inline model/compile/QtiAssetCompiler/XIncludeAdditionalAssetInjector.php:43]: This string is not existing as const anywhere yet?

**Review by bartlomiejmarszal — COMMENTED** [review]
- [x] New code is covered by tests (if applicable)
- [x] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [x] New code is not subject to concurrency issues (if applicable)
- [x] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful

**Review by mike-ionut-mihai-sandu-tao — APPROVED** [review]
- [ ] New code is covered by tests (if applicable)
- [x] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [ ] New code is respecting code style rules
- [x] New code is respecting best practices
- [x] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful

File: model/compile/QtiAssetCompiler/XIncludeAdditionalAssetInjector.php (hunk 8 above)
Comment (pnal) [reply model/compile/QtiAssetCompiler/XIncludeAdditionalAssetInjector.php:43]: For both constants answer is no

**Review by bartlomiejmarszal — APPROVED** [review]
LGTM

**Pull request merged**